lib/ansible/modules/source_control/git.py

```diff
@@ -432,6 +439,21 @@ def clone(git_path, module, repo, dest, remote, depth, version, bare,
                         "HEAD, branches, tags or in combination with refspec.")
     if reference:
         cmd.extend(['--reference', str(reference)])
+    separate_git_dir_fallback = False
+    if separate_git_dir is not None and bare:
+        module.warn("Ignoring separate_git_dir argument. "
+                    "Can not do bare clone with argument separate_git_dir.")
+    elif separate_git_dir:
```

Comment:
What if it already exists?

Reply:
If the repo already exists, the code will not invoke clone function, it will try to update the repo, so I think it's ok.